modules/ejb/src/main/java/org/jboss/weld/module/ejb/EjbSupportImpl.java

```diff
@@ -70,13 +73,25 @@ public void cleanup() {
         InternalEjbDescriptor<T> descriptor = InternalEjbDescriptor.of(d);
         EnhancedAnnotatedType<T> implementationClass = SessionBeans.getEjbImplementationClass(descriptor, manager, type);
 
-        AbstractInstantiator<T> instantiator = null;
+        Instantiator<T> instantiator = null;
         if (type.equals(implementationClass)) {
             instantiator = new DefaultInstantiator<T>(type, null, manager);
         } else {
             // Session bean subclassed by the EJB container
             instantiator = SubclassedComponentInstantiator.forSubclassedEjb(type, implementationClass, null, manager);
         }
+        InterceptionModel interceptionModel = manager.getInterceptorModelRegistry().get(type.slim());
+        if (interceptionModel != null) {
+            if (interceptionModel.hasExternalNonConstructorInterceptors()) {
+                instantiator = SubclassedComponentInstantiator
+                        .forInterceptedDecoratedBean(implementationClass, null, (AbstractInstantiator<T>) instantiator, manager);
```

Comment:
I think the `null` value here is OK as the MDB is not a bean but a mere injection target. Also a quick glance at the code shows it takes `null` into consideration and CI doesn't complain.